tests/test_mathematical.py

```diff
@@ -137,16 +138,13 @@ def _test_mathematical(self, name, dtype, lhs, rhs):
         else:
             result = getattr(dpnp, name)(a_dpnp, b_dpnp)
             expected = getattr(numpy, name)(a_np, b_np)
-            assert_allclose(result, expected, rtol=1e-6)
+            assert_dtype_allclose(result, expected, check_type=False)
```

Comment:
There are falls only for `test_arctan2` with `bool`  but this change disables type checking for all functions in `TestMathematical`. 
It would be better if you add default `check_type` argument to `_test_mathematical` and pass it it `test_arctan2`
```
self._test_mathematical("arctan2", dtype, lhs, rhs, check_type=False)
```